src/main/java/btdex/core/ContractState.java

```diff
@@ -423,12 +423,12 @@ private void buildHistory(Transaction[] txs) {
 					}
 				}
 				else if(rateHistory > 0L && marketHistory > 0L // we only want to register trades we know the price
-						&& !appendMessage.isText() && appendMessage.getMessage().length() == 64
-						&& appendMessage.getMessage().startsWith(Contracts.getContractTakeHash(type))) {
+						&& !appendMessage.isText() && jsonData.length() == 64
+						&& jsonData.startsWith(Contracts.getContractTakeHash(type))) {
 					// the take message (we are not so strict here as this is not vital information)
 					// so we can have false positives here, like multiple takes on the same order or invalid takes
 					// being account
-					byte []messageBytes = Hex.decode(appendMessage.getMessage());
+					byte []messageBytes = Hex.decode(jsonData);
```

Comment:
Naming the variable `jsonData` can be misleading here since it is not Json at this point but a hexadecimal.

Reply:
Agree. I have renamed the variable to messageString in next commit.